packages/gatsby-worker/src/child.ts

```diff
@@ -5,16 +5,33 @@ import {
   END,
   ERROR,
   RESULT,
+  CUSTOM_MESSAGE,
 } from "./types"
 import { isPromise } from "./utils"
 
+export interface IGatsbyWorkerMessenger<
+  MessagesFromParent = unknown,
+  MessagesFromChild = MessagesFromParent
+> {
+  onMessage: (listener: (msg: MessagesFromParent) => void) => void
+  sendMessage: (msg: MessagesFromChild) => void
+  messagingVersion: 1
+}
+
 /**
  * Used to check wether current context is executed in worker process
  */
 let isWorker = false
+let getMessenger = function <
+  MessagesFromParent = unknown,
+  MessagesFromChild = MessagesFromParent
+>(): IGatsbyWorkerMessenger<MessagesFromParent, MessagesFromChild> | undefined {
+  return undefined
+}
 
 if (process.send && process.env.GATSBY_WORKER_MODULE_PATH) {
   isWorker = true
+  const listeners: Array<(msg: any) => void> = []
```

Comment:
I think the `any` here could be a type union on multiple types like `ParentMessageUnion` or `MessagesFromParent`?

Reply:
Ideally yes, the problem for me was that `listeners` Set is outside of any generics so it doesn't have access to types defined by consumers (`MessagesFromParent` is one that would be used here). Maybe there is some TS way to do it, but I couldn't really think of how

Reply:
Yeah, with a generic this would be nicer and I don't know a good alternative to this top of my head. I'm fine with keeping it this way then for now, so not blocking